Backend/src/main/java/ro/uaic/info/romandec/services/ManuscriptService.java

```diff
@@ -2,44 +2,168 @@
 
 
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.io.FileSystemResource;
 import org.springframework.stereotype.Service;
 import org.springframework.web.multipart.MultipartFile;
+import ro.uaic.info.romandec.Request.SpecificManuscriptRequest;
+import ro.uaic.info.romandec.Response.ManuscriptDetailedResponse;
+import ro.uaic.info.romandec.Response.ManuscriptPreviewResponse;
 import ro.uaic.info.romandec.exceptions.InvalidDataException;
-import ro.uaic.info.romandec.exceptions.NoAvailableImageForAnnotator;
+import ro.uaic.info.romandec.exceptions.NoAvailableDataForGivenInput;
 import ro.uaic.info.romandec.models.Manuscript;
 import ro.uaic.info.romandec.models.ManuscriptMetadata;
+import ro.uaic.info.romandec.models.User;
 import ro.uaic.info.romandec.repository.ManuscriptMetadataRepository;
 import ro.uaic.info.romandec.repository.ManuscriptRepository;
+import ro.uaic.info.romandec.repository.UserRepository;
 
 import java.io.File;
 import java.io.IOException;
-import java.nio.charset.StandardCharsets;
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.Paths;
 import java.nio.file.StandardOpenOption;
-import java.util.List;
-import java.util.Objects;
+import java.util.*;
 
 @Service
 public class ManuscriptService {
 
     private final ManuscriptRepository manuscriptRepository;
+    private final UserRepository userRepository;
 
     private final ManuscriptMetadataRepository manuscriptMetadataRepository;
 
     private final String databaseDirectory = "../Database/";
 
     @Autowired
     public ManuscriptService(ManuscriptRepository manuscriptRepository,
-                             ManuscriptMetadataRepository manuscriptMetadataRepository)
+                             ManuscriptMetadataRepository manuscriptMetadataRepository,
+                             UserRepository userRepository)
     {
         this.manuscriptRepository = manuscriptRepository;
         this.manuscriptMetadataRepository = manuscriptMetadataRepository;
+        this.userRepository = userRepository;
     }
 
+    private String extractFilenameWithoutExtension(String filename)
+    {
+        if (filename == null || filename.isEmpty())
+        {
+            return filename;
+        }
+
+        int dotIndex = filename.lastIndexOf('.');
+
+        if (dotIndex <= 0)
+        {
+            return filename;
+        }
+
+        return filename.substring(0, dotIndex);
+    }
```

Comment:
Maybe make use of Java Paths
```suggestion
    private String extractFilenameWithoutExtension(String filename) {
    Path path = Paths.get(filename);
    return path.getName(path.getNameCount() - 1).toString();
}
```